baremetal/metal3machine_manager_test.go

```diff
@@ -4193,21 +4193,23 @@ var _ = Describe("Metal3Machine manager", func() {
 			expectedKcp:     true,
 			expectedKcpName: "kcp-test1",
 		}),
-		Entry("Should not find the expected kcp, kind is not correct", testCaseGetKubeadmControlPlaneName{
+		Entry("Should find the expected ControlPlane for any Kind/Provider", testCaseGetControlPlaneName{
 			Machine: &clusterv1.Machine{
 				ObjectMeta: metav1.ObjectMeta{
 					OwnerReferences: []metav1.OwnerReference{
 						{
 							APIVersion: controlplanev1.GroupVersion.String(),
-							Kind:       "kcp",
+							Kind:       "AnyControlPlane",
 							Name:       "test1",
 						},
 					},
 				},
 			},
-			expectError: true,
+			expectError:     false,
+			expectedKcp:     true,
+			expectedKcpName: "kcp-test1",
```

Comment:
Let's not re-use the KCP specific field here? We could introduce new ones in `testCaseGetControlPlaneName` for any other Controlplane provider then KCP and use them here perhaps to differentiate the two OR rename the fields to be generic?

Reply:
Ack I renamed all-the-things to be more generic :)